### ISSUE TITLE
Release macOS arm build of cinny-desktop + hombrew cask

### ISSUE THREAD

**Issue by jasikpark** [cinnyapp/cinny-desktop#62]
### Describe the problem

I'd like to install cinny desktop on my MacBook pro with an M1 chip via `brew install cinny`

### Describe the solution you'd like

Add `macos-arm` to the supported targets that `cinny-desktop` builds to + setup a homebrew cask tracking that repo

### Additional context

- https://docs.brew.sh/Cask-Cookbook
- https://github.com/tauri-apps/tauri-action
- https://github.com/tauri-apps/tauri-action/issues/243

**Comment by C0ffeeCode** [comment]
Maybe also release arm Windows builds.

**Comment by kfiven** [comment]
arm64 for linux as well. cinnyapp/cinny#553

**Comment by arjpar** [comment]
Any update on this @kfiven? I have a cask file prepared but Cinny desktop will have to be a universal binary and signed.

**Comment by kfiven** [comment]
@arjpar Last time I tried, couldn't build the universal binary using GitHub actions.

**Comment by arjpar** [comment]
@kfiven Apparently you can use a Github Action called `macos-universal-binary-action` to create and upload macOS universal binaries. This action takes several inputs such as owner, repository, tag, amd64, arm64 etc. You can find more details about this action on its Github Marketplace page [here](https://github.com/marketplace/actions/macos-universal-binary-action).

**Comment by solus-hq** [comment]
@arjpar Is it possible to somehow obtain a build or instruction to build for Mac with M1 processors?

**Comment by arjpar** [comment]
@solus-hq Are you talking about building on a Mac with Apple Silicon (M1, M2, etc)? For building I guess you would look at the GitHub action I linked in the comment [above](https://github.com/cinnyapp/cinny-desktop/issues/62#issuecomment-1477400714). I have no experience using GitHub actions though, so sorry I couldn't be of more help. I'm not sure if there's a GitHub action to create a Universal binary for Tauri applications.

**Comment by solus-hq** [comment]
@arjpar Yes, you're completely correct, I'm looking for some manual to build it for an M1 powered Mac. Have no experience with Github actions as well, but I guess I have to check if Tauri compiles on M1.

**Comment by arjpar** [comment]
@solus-hq You can compile Tauri apps to universal binaries. See [here](https://tauri.app/v1/guides/building/macos/).

**Comment by traktuner** [comment]
+1
please release cinny for ARM macOS. It's the only app which uses rosetta2 on my machine.

**Comment by ppkhoa** [comment]
+1 for Windows ARM64. As of now, there's no Matrix client desktop app for Windows arm64 (not counting web app through arm64 browser)

**Comment by arjpar** [comment]
@ppkhoa You should probably open a separate issue for Windows on ARM64 if there isn't an open issue already.

**Comment by ghost** [comment]
I've seen there is even a macos-aarch64 workflow which is just commented out.

It might be interesting to know what the issue was there

**Comment by bernardoaraujor** [comment]
any updates?

**Comment by leisefuxX** [comment]
*bump* possible?

**Comment by Neved4** [comment]
Made my own tap 🎉
```sh
brew install neved4/tap/cinny
```

The above should just work. Ideally, Cinny would roll its own tap upstream, like: [homebrew-tap/cinny.rb].

To do it in the official Homebrew repos, since [_Apple Silicon requires signed code_] you'll likely have to generate a certificate, add it to your Github secrets, then modify [tauri.yml] as per [tauri.app/sign-macos/#example]. Cheers 🤍

[_Apple Silicon requires signed code_]: https://eclecticlight.co/2020/08/22/apple-silicon-macs-will-require-signed-code/
[tauri.yml]: https://github.com/cinnyapp/cinny-desktop/blob/main/.github/workflows/tauri.yml#L152-L155
[tauri.app/sign-macos/#example]: https://tauri.app/v1/guides/distribution/sign-macos/#example
[homebrew-tap/cinny.rb]: https://github.com/Neved4/homebrew-tap/blob/main/Casks/cinny.rb

**Comment by InvitedToHell** [comment]
Any updates on making this official?

**Comment by trymeouteh** [comment]
Would like to see a homebrew package as well
